scripts/hudson/narayana.sh

```diff
@@ -385,6 +366,28 @@ function clone_as {
 function build_as {
   echo "Building JBoss EAP/WildFly"
 
+  OVERRIDE_LRA_VERSION=""
+  
+  echo "Checking if need LRA PR"
+  if [ -n "$LRA_BRANCH" ]; then
+    echo "Building LRA PR"
+    if [ -d lra ]; then
+      rm -rf lra
+    fi
+    git clone https://github.com/jbosstm/lra.git -o jbosstm
+    [ $? -eq 0 ] || fatal "git clone https://github.com/jbosstm/lra.git failed"
+    cd lra
+    git fetch jbosstm +refs/pull/*/head:refs/remotes/jbosstm/pull/*/head
+    [ $? -eq 0 ] || fatal "git fetch of pulls failed"
+    git checkout $LRA_BRANCH
+    [ $? -eq 0 ] || fatal "git fetch of pull branch failed"
+    cd ../
+    ./build.sh -f lra/pom.xml -B clean install
+    [ $? -eq 0 ] || fatal "Build of LRA failed"
+    LRA_VERSION=`awk '/lra-parent/ { while(!/<version>/) {getline;} print; }' lra/pom.xml | cut -d \< -f 2|cut -d \> -f 2`
+    OVERRIDE_LRA_VERSION=" -Dversion.org.jboss.narayana.lra=${LRA_VERSION} "
+  fi
+  
```

Comment:
I was wondering if it makes sense to build WildFly with the main branch of lra in case `LRA_BRANCH` is not provided. WDYT @marcosgopen?

Reply:
I think I sent my previous message to quickly. What's the reason to build WildFly with a different LRA version if the Narayana repo won't test LRA at all? @marcosgopen

Reply:
This is an excellent question @jmfinelli. We are at a stage where no lra artifacts are present in maven central. Those artifacts are needed for Wildfly to be built. So at the moment I am building lra for having the artifacts needed by the  WildFly build process.

Reply:
When lra artifacts are available on nexus we can rely on WildFly to be able to retrieve the lra artifacts by itself, see  https://github.com/jbosstm/jboss-as/pull/92

Reply:
So, if I understand correctly, this script is a temporary patch to build WildFly while we are still in the middle of producing the first (indipendent) LRA artifact and update WildFly's bom. Is my understanding correct?